File: expo-sdk/ios/MdocDataTransferEvent.swift

enum MdocDataTransferEvent {
    case ON_REQUEST_RECEIVED
    case ON_RESPONSE_SENT
    case ON_ERROR

    var description : String {
        switch self {
        case .ON_REQUEST_RECEIVED: "onRequestReceived"
        case .ON_RESPONSE_SENT: "onResponseSent"
        case .ON_ERROR: "onError"
        }
    }
}

File: expo-sdk/src/MdocDataTransferEvent.ts
export enum MdocDataTransferEvent {
  OnRequestReceived = 'onRequestReceived',
  OnResponseSent = 'onResponseSent',
  OnError = 'onError',
}

export type OnResponseSentPayload = null

export type OnRequestReceivedEventPayload = {
  deviceRequest: string
  sessionTranscript: string
}

export type OnErrorPayload = {
  error: string
}
